### PR TITLE
add lockfile bump and release notes draft tools

### PR DESCRIPTION
**Pull request by jsvd**
adds two scripts to `tools/release`:
- `bump_plugin_versions.rb`: auto generates a PR with updated lock file [example](https://github.com/elastic/logstash/pull/10262)
- `generate_release_notes.rb`: auto generate a PR with a release notes draft for a new release [example](https://github.com/elastic/logstash/pull/10261)

### CODE REVIEW

File: tools/release/bump_plugin_versions.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+puts `git diff Gemfile.jruby-2.3.lock.release`
+
+puts "Creating commit.."
+
+branch_name = "update_lock_#{Time.now.to_i}"
+`git checkout -b #{branch_name}`
+`git commit Gemfile.jruby-2.3.lock.release -m "Update #{allow_bump_for} plugin versions in gemfile lock"`
+
+puts "Pushing commit.."
```
Comment (colinsurprenant) [inline tools/release/bump_plugin_versions.rb]: Are we sure we want to automate the branch push? Will that be part of an automated workflow or is that something we must manually run, in which case maybe a manual validation step might be safer (maybe using a confirmation prompt?)

File: tools/release/generate_release_notes.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  report << ""
+  start_changelog_file.unlink
+  end_changelog_file.unlink
+end
+
+release_notes.insert(release_notes_entry_index, report.join("\n").gsub(/\n{3,}/, "\n\n"))
+
+IO.write(RELEASE_NOTES_PATH, release_notes.join("\n"))
+
+puts "Creating commit.."
```
Comment (colinsurprenant) [inline tools/release/generate_release_notes.rb]: same comment here, should we add a validation step to make sure we want to proceed with the commit and push?

File: tools/release/bump_plugin_versions.rb (hunk 1 above)
Comment (jsvd) [reply tools/release/bump_plugin_versions.rb]: it doesn't push directly to the branch, it creates a pull request, so I think it's safe enough?

Comment (colinsurprenant) [reply tools/release/bump_plugin_versions.rb]: it does a commit and a branch push - it is just a bummer if you realized you made a mistake and have to revert the commit where a simple "this is what I am about to do; you asked to do [...[ here are the diffs [...],  are you sure you want proceed with the commit and push?"

Comment (jsvd) [reply tools/release/bump_plugin_versions.rb]: the purpose of these tools is to be run through CI, so we don't have a place to have this interaction through jenkins.

As for realizing mistakes and fixing them, it's a throwaway branch, and you can always take the commit and work on top of it and push that instead on a separate branch.

Comment (jsvd) [reply tools/release/bump_plugin_versions.rb]: that said, +1 on adding flags to not create a PR (e.g. `--[no-]create-pr`) but I don't think we need to do it in this PR

**Comment by colinsurprenant** [comment]
LGTM - not tested, low risk, these are dev tools and we can easily iterate on if we discover problems.

**Pull request closed without merging**